### PR TITLE
add UpdateCurrentApplicationsCount

### PR DESCRIPTION
**Pull request by alesyalis**

### CODE REVIEW

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IProjectRepository.cs [hunk 1]
```diff
@@ -8,5 +8,7 @@ namespace Recrutify.DataAccess.Repositories.Abstract
     public interface IProjectRepository : IBaseRepository<Project>
     {
         Task<IEnumerable<ProjectPrimarySkill>> GetPrimarySkills(Guid id);
+
+        Task UpdateCurrentApplicationsCount(Guid id);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/Abstract/IProjectRepository.cs:12]: IncrementCurrentApplicationsCountAsync

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 2]
```diff
@@ -24,5 +24,13 @@ public Task<IEnumerable<ProjectPrimarySkill>> GetPrimarySkills(Guid id)
              .Project(p => p.PrimarySkills)
              .FirstOrDefaultAsync();
         }
+
+        public Task UpdateCurrentApplicationsCount(Guid id)
+        {
+            var filter = _filterBuilder.Eq(x => x.Id, id);
+            var updateBuilder = Builders<Project>.Update;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:31]: you can not write separate var updateBuilder, you use it once, so can just write Builders<Project>.Update.... in next line

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 3]
```diff
@@ -24,5 +24,13 @@ public Task<IEnumerable<ProjectPrimarySkill>> GetPrimarySkills(Guid id)
...
+            var updateDefinition = updateBuilder.Inc("CurrentApplicationsCount", 1);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:32]: use linq to define field to update

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 4]
```diff
@@ -17,14 +17,16 @@ namespace Recrutify.Services.Services
     public class CandidateService : ICandidateService
     {
         private readonly ICandidateRepository _candidateRepository;
+        private readonly IProjectRepository _projectRepository;
         private readonly IMapper _mapper;
         private readonly IValidator<ProjectResult> _validator;
 
-        public CandidateService(ICandidateRepository candidateRepository, IMapper mapper, IValidator<ProjectResult> validator)
+        public CandidateService(ICandidateRepository candidateRepository, IMapper mapper, IValidator<ProjectResult> validator, IProjectRepository projectRepository)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:24]: pass IProjectService not repository

Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs (hunk 2 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/IProjectRepository.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/Abstract/IProjectRepository.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs (hunk 3 above)
Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs]: done

File: Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs [hunk 5]
```diff
@@ -24,5 +24,12 @@ public Task<IEnumerable<ProjectPrimarySkill>> GetPrimarySkills(Guid id)
              .Project(p => p.PrimarySkills)
              .FirstOrDefaultAsync();
         }
+
+        public Task IncrementCurrentApplicationsCountAsync(Guid id)
+        {
+            var filter = _filterBuilder.Eq(x => x.Id, id);
+            var updateBuilder = Builders<Project>.Update.Inc(p => p.CurrentApplicationsCount, 1);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:31]: updateDefinition

Comment (alesyalis) [reply Recrutify/Recrutify.DataAccess/Repositories/ProjectRepository.cs:31]: done

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**